### PR TITLE
add fineli example data

### PR DESCRIPTION
**Pull request by bigdatabaracus**

### CODE REVIEW

**Comment by bigdatabaracus** [comment]
One thing to discuss here: should the fineli data ~ 100M be added into the repository or should it be loaded on installation as it is done now (as a part of `npm install`)

**Comment by bigdatabaracus** [comment]
@Kaali Could you take a look at this? Thanks ☺️

**Comment by Kaali** [comment]
Could you clean the commit message and run the code through prettier as the first step. Still reading and testing the rest.

File: data/tokenScores.js [hunk 1]
```diff
@@ -43,39 +43,44 @@ const createSGApiCall = (wlu, language) => (text, accum, authToken) => {
...
-        return previousPromise.then(accum => {
-          bar.tick();
-          return apiCall(text, accum, authToken);
+    const identity = new SgGrpc.speechgrinder.sgapi.v1.Identity(sgApiUrl, creds);
+    const wlu = new SgGrpc.speechgrinder.sgapi.v1.Wlu(sgApiUrl, creds);
+    const apiCall = createSGApiCall(wlu, language);
+    const deviceId = "fineli-lemmatizer";
+    identity.login({ appId, deviceId }, (err, response) => {
+      if (err) {
+        throw new Error(err.message);
```
Comment (langma) [inline data/tokenScores.js]: `reject(err);`?

File: data/tokenScores.js [hunk 2]
```diff
@@ -43,39 +43,44 @@ const createSGApiCall = (wlu, language) => (text, accum, authToken) => {
...
+            bar.tick();
+            return apiCall(text, accum, authToken);
+          });
+        }, Promise.resolve([]))
+        .then(t => {
+          resolve(t);
+        })
+        .catch(err => {
+          console.log(
+            ` Skipping data index creation for language ${language}! (not supported by give application ID)`
```
Comment (langma) [inline data/tokenScores.js]: Could also log the invalid appId here

**Comment by bigdatabaracus** [comment]
Thank you @langma for the comments. Both details you point out in your review are deliberately like they are now (whether reasonable is of course up for debate). This is motivated due to the data install script preprocessing both English and Finnish although some (and probably most) appids support only one or the other. Now the problem from the point of view of running the script is that currently the SG API does not support, as far as I am aware, querying a priori what languages are supported. Therefore what the script does is it tries to run indexing with both languages and then just skip the language that is not supported (and produces and error) by the appid. Perhaps a more elegant implementation could be provided once SG API supported providing supported languages?

File: data/tokenScores.js (hunk 1 above)
Comment (bigdatabaracus) [reply data/tokenScores.js]: This error occurs if app id is invalid. Here we want to output the error and halt the script. Should we convey the error to `reject`, the script would as it currently is try to continue processing with the next language.

File: data/tokenScores.js (hunk 2 above)
Comment (bigdatabaracus) [reply data/tokenScores.js]: This error should occur when the language is invalid. Here we want to continue to the script by trying out the next language. If the appid is invalid the script should have already halted already earlier on line 63 (previous comment).

File: data/tokenScores.js (hunk 1 above)
Comment (langma) [reply data/tokenScores.js]: Yes, if login fails, it should not continue running. This code now does as many logins as there are languages? The login phase should be handled before this anyway, as the login result is reusable. Also I'm not too sure about the device id, but ATM we do require it in the login phase.

Comment (bigdatabaracus) [reply data/tokenScores.js]: Yes, good point @langma. I've now updated the code to have the login be called before and only once. See if it's better now!

**Comment by bigdatabaracus** [comment]
add fineli example data

**Pull request merged**